### PR TITLE
APR-2950 functions data structure enhancements error handling, logging and enqueuing files call idempotent

### PR DESCRIPTION
**Pull request by raviudari01**
Functions data structure enhancements error handling and logging, and ordering enqueuing files call to make it idempotent.

### CODE REVIEW

File: src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs [hunk 1]
```diff
@@ -211,39 +210,47 @@ public async Task<List<Guid>> GetApplicationsToExtract(DateTime executionDateTim
...
+                    await ExtractPartnershipAnswers(applicationId, answers);
+                }
             }
-            //Extract QuestionId Y0-110 answer even if it is inactive, there is an issue setting answer active to false while adding orgnisation type answer to the tabular list as the last entry
-            var hasPartnershipAnswersExtracted = answers.Any(answer => answer.QuestionId == QuestionIdPartnership);
-            if(!hasPartnershipAnswersExtracted)
+            catch (Exception ex)
             {
-                await ExtractPartnershipAnswers(applicationId, answers);
+                _logger.LogError(ex, $"Unable to extract answers for application {applicationId}");
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs:243]: Remove try catch and use log as discussed before. Or throw the exception.

File: src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs [hunk 2]
```diff
@@ -211,39 +210,47 @@ public async Task<List<Guid>> GetApplicationsToExtract(DateTime executionDateTim
...
             }
-
             return answers;
         }
 
         private async Task ExtractPartnershipAnswers(Guid applicationId, List<SubmittedApplicationAnswer> answers)
         {
             try
             {
+                _logger.LogInformation($"Extract Partnership answers for application {applicationId}");
```
Comment (DevenKShah) [inline src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs:252]: Same here... remove try catch or throw exception

**Review by DevenKShah — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs (hunk 2 above)
Comment (raviudari01) [reply src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs:253]: Addressed with throw exception

File: src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs (hunk 1 above)
Comment (raviudari01) [reply src/SFA.DAS.Roatp.Functions/ApplicationExtract.cs:243]: Addressed with throw exception

**Review by DevenKShah — APPROVED** [review]

**Pull request merged**